share/functions/__fish_inc_dec_number_under_cursor.fish

```diff
@@ -0,0 +1,90 @@
+# Usage:
+    # This function can be bound to key shortcuts (like Ctrl+A for increase and Ctrl+X for decrease) in fish shell to modify numbers quickly under the cursor.
+    # Key bindings for increasing and decreasing numbers in insert/vi mode mode.
+    # bind -M insert \ca '__fish_inc_dec_number_under_cursor increase'
+    # bind -M insert \cx '__fish_inc_dec_number_under_cursor decrease'
+
+function __fish_inc_dec_number_under_cursor
+    # Capture the current command line buffer and cursor position.
+    set current_buffer (commandline -b)  # Get the entire current command line as a string.
+    set cursor_position (commandline -C)  # Get the current cursor position.
+    set original_cursor_position (commandline -C)  # Store the original cursor position for restoring later.
+    
+    # Get the character directly under the cursor.
+    set cursor_character (echo $current_buffer | cut -c (math $cursor_position + 1))
+
+    # If the character under the cursor is not a digit, repaint the command line and exit.
+    if not string match -qr '^[0-9]+$' $cursor_character
+        commandline -f repaint
+        return
+    end
+
+    # Capture the character at the current cursor position (for safety, 2>/dev/null hides errors if position is invalid).
+    set character (echo "$current_buffer" | cut -c $cursor_position 2>/dev/null)
+
+    # Move the cursor backwards to find the start of the number under the cursor.
+    while string match -qr '^[0-9]+$' $character  # Check if the character is a digit.
+        set cursor_position (math $cursor_position - 1)  # Move the cursor one position left.
+        set character (echo "$current_buffer" | cut -c $cursor_position 2>/dev/null)  # Update character.
+    end
+
+    # If the character is '-', keep cursor position (for negative numbers).
+    if not string match -qr '-' $character
```

Comment:
this seems broken (try running `string match -qr - -`). Need to add a `--` I think